vault/ui_custom_messages/entry.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package uicustommessages
+
+import (
+	"errors"
+
+	"github.com/hashicorp/go-uuid"
+)
+
+// Entry is a struct that contains a map of custom message ID to Message struct
+// that can be marshalled and unmarshalled using the JSON encoding to/from a
+// slice of byte to translate to a logical.StorageEntry struct. An Entry
+// consists of data for a particular namespace.
+type Entry struct {
+	// Messages is the map of custom message ID to Message struct.
+	Messages map[string]Message `json:"messages"`
+}
+
+// FindMessages searches through all of the custom messages in the receiver
+// Entry struct and only returns those that match the criteria set in the
+// provided FindFilter struct.
+func (e *Entry) FindMessages(filter FindFilter) []Message {
+	result := make([]Message, 0)
+
+	for _, message := range e.Messages {
+		if message.Matches(filter) {
+			result = append(result, message)
+		}
+	}
+
+	return result
+}
+
+// CreateMessage adds a custom message in the receiver Entry struct using the
+// provided Message struct to populate its properties. If the either the
+// start/end times are invalid or the maximum number of messages already exists,
+// then the message is not added.
+func (e *Entry) CreateMessage(message *Message) error {
+	uuid, err := uuid.GenerateUUID()
+	if err != nil {
+		return err
+	}
+
+	if !message.ValidateStartAndEndTimes() {
```

Comment:
Given that we use this 'on' a message struct, I wonder if we should rename this to e.g. `message.HasValidStartAndEndTimes`? Same for `message.HasValidMessageType`.

To me, `ValidateStartAndEndTimes(message)` and `message.HasValidStartAndEndtimes` feel best, so some food for thought